src/views/view_sycl.hpp

```diff
@@ -57,98 +57,58 @@ struct VectorView<
   using container_t =
       typename codeplay_policy::template placeholder_accessor_t<scalar_t>;
   using self_t = VectorView<scalar_t, container_t, index_t, increment_t>;
+  const index_t size_;
+  const index_t disp_;
+  const increment_t strd_;  // never size_t, because it could be negative
   container_t data_;
-  index_t size_data_;
-  index_t size_;
-  index_t disp_;
-  increment_t strd_;  // never size_t, because it could negative
 
   using value_t = scalar_t;
 
+  static SYCL_BLAS_INLINE const index_t calc_size(container_t &data,
```

Comment:
What is this function for? Could it please have a documentation comment.

Reply:
I'm also confused by the negative stride. Isn't increment_t supposed to be always >0 ?

Reply:
@adam added. @przemektmalon  no it can be negative that is the reason every index and increment parameter in BLAS is signed.

Reply:
Ah ok, I misinterpreted the comment at line 62